components/BlurhashImage.vue

```diff
@@ -57,81 +62,82 @@ export default Vue.extend({
       image: ''
     };
   },
-  computed: {
-    checkImageHash(): boolean {
+  mounted(): void {
+    const img = this.$refs.img as HTMLElement;
+    // We don't pass the item itself as we already did all the tags checking in this component,
+    // so doing it again in the mixin is useless.
+    this.image = this.getImageUrlForElement(this.type, {
+      itemId: this.item.Id,
+      element: img
+    });
+  },
+  methods: {
+    ...mapActions('snackbar', ['pushSnackbarMessage']),
+    isValidTag(): boolean {
       if (
-        (this.imageType === ImageType.Primary &&
-          this.item.ImageBlurHashes?.Primary &&
-          this.item.ImageTags?.Primary) ||
-        (this.imageType === ImageType.Backdrop &&
-          this.item.ImageBlurHashes?.Backdrop &&
-          this.item.ImageTags &&
-          this.item.BackdropImageTags) ||
-        (this.imageType === ImageType.Thumb &&
-          this.item.ImageBlurHashes?.Thumb &&
-          this.item.ImageTags?.Thumb)
+        (this.item?.ImageTags &&
+          Object.prototype.hasOwnProperty.call(
+            this.item?.ImageTags,
+            this.type
+          )) ||
+        (this.type === ImageType.Backdrop &&
+          this.item?.BackdropImageTags &&
+          this.item?.BackdropImageTags.length > 0)
       ) {
         return true;
+      } else {
+        this.onError();
+        return false;
       }
-      return false;
-    },
-    getImageHash(): string {
-      switch (this.imageType) {
-        case ImageType.Primary:
-          if (
-            this.item?.ImageTags?.Primary &&
-            this.item.ImageBlurHashes?.Primary?.[this.item?.ImageTags?.Primary]
-          ) {
-            return this.item.ImageBlurHashes?.Primary?.[
-              this.item?.ImageTags?.Primary
-            ];
-          }
-          break;
-        case ImageType.Thumb:
-          if (
-            this.item?.ImageTags?.Thumb &&
-            this.item.ImageBlurHashes?.Thumb?.[this.item?.ImageTags?.Thumb]
-          ) {
-            return this.item.ImageBlurHashes?.Thumb?.[
-              this.item?.ImageTags?.Thumb
-            ];
-          }
-        // eslint-disable-next-line no-fallthrough
-        case ImageType.Backdrop || ImageType.Thumb:
-          if (
-            this.item?.BackdropImageTags?.[0] &&
-            this.item?.ImageBlurHashes?.Backdrop
-          ) {
-            return this.item?.ImageBlurHashes?.Backdrop?.[
-              this.item?.BackdropImageTags?.[0]
-            ];
-          }
-      }
-      return '';
     },
-    checkImage(): boolean {
+    isValidBlurhash(): boolean {
```

Comment:
This doesn't check if the Blurhash is actually valid, just if it exists in the item.

We should perform a check via Blurhash's `isBlurhashValid` function, too.

Reply:
@MrTimscampi That's actually done in the worker with a try/catch (the function performs a halfway decode and it reported false positives for me, so I think doing the decode directly and catching when needing is our best bet)

Reply:
> it reported false positives

This might be a bug in our implementation server-side. I know we at least have a color channel issue, since results are pretty different between our custom C# library and Blurhash's reference one in terms of color.

But that sounds good for now.